contrib/dracut/90zfs/zfs-generator.sh.in

```diff
@@ -15,8 +15,10 @@ type getarg >/dev/null 2>&1 || . "$dracutlib"
 [ "${root##zfs:}" = "${root}" -a "${root##ZFS=}" = "${root}" -a "$rootfstype" != "zfs" ] && exit 0
 
 rootfstype=zfs
-if echo "${rootflags}" | grep -q zfsutil ; then
+if echo "${rootflags}" | grep -Eq '^zfsutil$|^zfsutil,|,zfsutil$|,zfsutil,' ; then
     true
+elif test -n "${rootflags}" ; then
+    rootflags="zfsutil,${rootflags}"
 else
     rootflags=zfsutil
 fi
```

Comment:
Sorry, you're right about the generator script, but I realized there is one way that we can do the auto boot.  If root is unset in the kernel command line and the initramfs, systemd-fstab-generator will not create sysroot.mount.  If zfs-generator bails when it sees "zfs:AUTO", there will no sysroot.mount, so mount-zfs.sh will run and do the auto mount.  Just need this in zfs-generator.sh:

```bash
if [ "${root}" = "zfs:AUTO" ] ; then
  exit 0
fi
```

I just booted using this patch set and that change, and it works with root unspecified.